source/slang/slang-language-server.h

```diff
@@ -100,7 +100,8 @@ class LanguageServer
     Slang::InlayHintOptions m_inlayHintOptions;
     bool m_quit = false;
     List<LanguageServerProtocol::WorkspaceFolder> m_workspaceFolders;
-
+    RttiTypeFuncsMap m_typeMap;
```

Comment:
This does not need to be initialized? I wonder when do we actually register new entries into the map.

Reply:
I think this is initialized in init on the LanguageServer. 

I should have elaborated more on the need for the map, which is that the Rtti assorted tools only worked with types that were "known" by the RttiUtil which didn't include JSONValue. RttiUtil is defined in core. Having JSONValue as a *native* value was useful to be able to make the distinction between null and a string value which was at best somewhat contrived with a slice or worse a string. 

For the current Language server, the use of types "outside" of RttiUtil worked ok. It didn't work though with the List of JSONValue I was using inside the native representation of the SourceMap type. 

Short version - if it wasn't initialized it would *currently at least* work ok with language server. It is initialized I think in init, so it should be covered.

On having to add the type, there was also the benefit of having a map to do lookup as opposed to a whole bunch of cases, and the implementation caches the results of RttiUtil which is perhaps a *small* benefit.